apps/sms/services/js/moz_mobile_connections/moz_mobile_connections_shim.js

```diff
@@ -0,0 +1,104 @@
+/* global BridgeServiceMixin, BroadcastChannel, MozSettingsShim */
+
+/* exported MozMobileConnectionsShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for mozMobileConnections API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'moz-mobile-connections-shim';
+
+/**
+ * Array of method names that need to be exposed for API shim.
+ * @type {Array.<string>}
+ */
+const METHODS = Object.freeze(['getServiceIdByIccId', 'switchMmsSimHandler']);
+
+var mozMobileConnections = null;
+
+function setMmsSimServiceId(id) {
+  // DSDS: mms & data are both necessary for connection switch.
+  MozSettingsShim.set({
+    'ril.mms.defaultServiceId': id,
+    'ril.data.defaultServiceId': id
+  });
+}
+
+var MozMobileConnectionsShim = {
+  init(conns) {
+    if (!conns) {
+      console.error('Invalid settings for shim initialization');
+      return;
+    }
+    mozMobileConnections = conns;
+
+    this.initService(
+      new BroadcastChannel('moz-mobile-connections-shim-channel')
```

Comment:
And for the appInstanceId - yeah we will need this eventually if we use BC for communication. This goal of this patch is simply make this shim works first, and I'll add it in the final version.